Azure Terraform Config and README updates

jasonhawkharris opened the pull request:
- add terraform configuration for provisioning AKS cluster
- add/update readme's

jasonhawkharris committed:
add azure tf config

jasonhawkharris committed:
add tfvars vars; remove housekeeping stuff

jasonhawkharris committed:
add note for configuring kubectl with az

jasonhawkharris committed:
Formatting

jasonhawkharris committed:
jhh/azure-tf-config

jasonhawkharris committed:
README final drafts; add configurable variables

jasonhawkharris committed:
README readability edits

jasonhawkharris committed:
bash for kubectl configuration for az on its own line

jasonhawkharris committed:
README edits

jasonhawkharris committed:
correct header sizes in root README

beyang commented on README.md:
```diff
@@ -1,25 +1,11 @@
-### Deploy a Kubernetes Cluster
+## Deploy a Kubernetes Cluster with Terraform
+These Terraform configurations deploy Kubernetes clusters on AWS (EKS), GCP (GKE), and Azure (AKS) with ease. Just specify a few variables and watch your cluster come to life!
```
Write in a more matter of fact, documentation-style tone

jasonhawkharris committed:
Revise readme with more professional tone

beyang commented on README.md:
```diff
@@ -1,25 +1,11 @@
-### Deploy a Kubernetes Cluster
+## Deploy a Kubernetes Cluster with Terraform
+These Terraform configurations automate the provisioning of Kubernetes clusters on AWS (EKS), GCP (GKE), or Azure (AKS). With a few variable inputs, you can launch a fully managed Kubernetes cluster on the cloud provider of your choice.
```
```suggestion
This repository provides examples of provisioning Kubernetes clusters on major cloud providers using Terraform. These should be treated as illustrative examples, not production-ready infrastructure configuration.
```

beyang commented on README.md:
```diff
@@ -1,25 +1,11 @@
-Pick the platform you want to deploy your cluster to, and follow the instructions in the README of the corresponding directory.
+### Get Started
+- Install [terraform](https://developer.hashicorp.com/terraform/downloads)
+- Pick your cloud provider: AWS, GCP, or Azure
+- Navigate to the corresponding directory
```
```suggestion
- `cd` into the directory corresponding to your cloud provider (AWS, GCP, or Azure)
```

beyang commented on aws/README.md:
```diff
@@ -6,16 +6,31 @@
 ### Set Your Variables
-1. Clone this repository to your local machine.
-2. Uncomment and replace variables in `terraform.tfvars` with the desired values. Check `variables.tf` for default values.
+1. Fork and clone this repository to your local machine.
```
Haven't we already cloned at this point (should be handled by root README?)

jasonhawkharris replied:
I chose to clarify this step rather than removing it altogether. "If you haven't already done so,...". That way we account for the user who _has_ already cloned, and the one who's still perusing the repo.

beyang commented on aws/README.md:
```diff
@@ -6,16 +6,31 @@
-1. Clone this repository to your local machine.
-2. Uncomment and replace variables in `terraform.tfvars` with the desired values. Check `variables.tf` for default values.
+1. Fork and clone this repository to your local machine.
+2. `cd` into `./aws`, then uncomment and replace variables in `terraform.tfvars` with the desired values. Check `variables.tf` for default values.
```
Safe to assume we're already in `aws` at this point?

jasonhawkharris replied:
Yup you're right.

beyang commented on gcp/README.md:
```diff
@@ -2,26 +2,44 @@
 
-Congrats. You have a Kubernetes cluster. Next, [it's time to deploy Sourcegraph](https://docs.sourcegraph.com/admin/deploy/kubernetes/configure).
+It will take a few minutes to deploy the cluster. When finished, you will be greeted with a success message.
+Congratulations! You have a Kubernetes cluster. Next, it's time to [deploy Sourcegraph](https://docs.sourcegraph.com/admin/deploy/kubernetes/configure).
```
```suggestion
Next, [deploy Sourcegraph](https://docs.sourcegraph.com/admin/deploy/kubernetes/configure) onto the cluster you just provisioned.
```

beyang reviewed (approved):
LGTM pending changes

jasonhawkharris committed:
Add note about the nature of the project

Co-authored-by: Beyang Liu <beyang@sourcegraph.com>

jasonhawkharris committed:
Update README.md

Co-authored-by: Beyang Liu <beyang@sourcegraph.com>

jasonhawkharris committed:
Remove hype language gcp/README.md

Co-authored-by: Beyang Liu <beyang@sourcegraph.com>

jasonhawkharris committed:
Add clarification about forking/cloning in READMEs
- Rather than deleting the "fork and clone" step
  completely, I chose to modify the comment to account for
  both the user that has already forked and cloned
  the repo, and the user who hasn't yet.

jasonhawkharris committed:
Nvm. Removed the 'fork and clone' step

beyang commented on README.md:
```diff
@@ -1,25 +1,10 @@
-
-4. Run `kubectl config current-context` to see the current context
-5. (Optionally) Run `kubectl cluster-info` to see information about the current cluster
+That's it. Next up, [deploy Sourcegraph](https://docs.sourcegraph.com/admin/deploy/kubernetes).
```
```suggest
Next, [deploy Sourcegraph](https://docs.sourcegraph.com/admin/deploy/kubernetes) onto the cluster you just provisioned.
```

Replace market-y, colloquial language with matter-of-fact doc-style language.

jasonhawkharris replied:
Pushed this exact change to main.

beyang commented on README.md:
```diff
@@ -1,25 +1,10 @@
+### Get Started
+- Install [terraform](https://developer.hashicorp.com/terraform/downloads)
+- `cd` into the directory corresponding to your cloud provider (AWS, GCP, or Azure)
+- Follow the instructions in the corresponding README
```
Replace these 2 bullets with one: "`cd` into the directory corresponding to your cloud provider (`aws`, `gcp`, or `azure`) and follow the instructions in the README in that directory."

jasonhawkharris replied:
Pushed this exact change to main.

beyang commented on aws/README.md:
```diff
@@ -6,16 +6,30 @@
 
-Congrats! you have a Kubernetes cluster! Next, [it's time to deploy Sourcegraph](https://docs.sourcegrapn.com/admin/deploy/kubernetes/configure)
+It will take a few minutes to deploy the cluster. When finished, you will be greeted with a success message.
+Congratulations! You have a Kubernetes cluster. Next, it's time to [deploy Sourcegraph](https://docs.sourcegraph.com/admin/deploy/kubernetes/configure).
```
Remove "Congratulations!" and "it's time to" (see previous comment about avoiding overly colloquial/marketing-style language)

jasonhawkharris replied:
Pushed this exact change to main.

beyang commented on aws/README.md:
```diff
@@ -6,16 +6,30 @@
+Congratulations! You have a Kubernetes cluster. Next, it's time to [deploy Sourcegraph](https://docs.sourcegraph.com/admin/deploy/kubernetes/configure).
+
+## Manage Your Cluster with Kubectl
+To manage your cluster, you can use the `kubectl` [command line tool](https://kubernetes.io/docs/tasks/tools/), which you'll need to configure cluster access for.
```
Worth putting this in the "Prerequesites" section?

jasonhawkharris replied:
I thought about it, but while I was testing these configs, I found that I really didn't configure kubectl until I'd already deployed the cluster. So I thought it would make more sense sequentially to place it here.

jasonhawkharris replied:
Throughout the process of testing these configs, I found that this step was never taken before, but always after I provisioned the cluster. IMO, it is better-suited where it is.

beyang replied:
ah, fair
